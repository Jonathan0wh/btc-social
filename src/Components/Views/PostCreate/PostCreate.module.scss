.container {
  margin-top: 10px;
  margin-bottom: 10px;
}

.textarea {
  height: 150px !important;
}
