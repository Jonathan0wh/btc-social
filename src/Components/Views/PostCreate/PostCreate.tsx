import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';
import Swal from 'sweetalert2';
import {
  Container,
  Row,
  Col,
  Form,
  FormGroup,
  Label,
  Input,
  Button,
  Spinner
} from 'reactstrap';

import { API_BASE_URL } from 'config/constant';
import styles from './PostCreate.module.scss';

const PostCreate = () => {
  const [title, setTitle] = useState('');
  const [body, setBody] = useState('');
  const [userId, setUserId] = useState(0);
  const [submitting, setSubmitting] = useState(false);
  let history = useHistory();

  return (
    <Container className={styles.container}>
      <Row>
        <Col>
          <h3>Create a Post</h3>
          <Form>
            <FormGroup>
              <Label for="post_title">Title</Label>
              <Input
                id="post_title"
                type="text"
                onChange={event => setTitle(event.target.value)}
              />
            </FormGroup>

            <FormGroup>
              <Label for="post_body">Content</Label>
              <Input
                id="post_body"
                type="textarea"
                onChange={event => setBody(event.target.value)}
                className={styles.textarea}
              />
            </FormGroup>

            <FormGroup>
              <Label for="post_userid">My user ID</Label>
              <Input
                id="post_userid"
                type="number"
                onChange={event => setUserId(parseInt(event.target.value))}
              />
            </FormGroup>

            <Button
              onClick={event => {
                event.preventDefault();
                setSubmitting(true);
                axios
                  .post(API_BASE_URL + 'posts', { title, body, userId })
                  .then(() => {
                    setSubmitting(false);
                    Swal.fire({
                      text: 'Post Success',
                      icon: 'success',
                      onClose: () => {
                        history.push('/');
                      }
                    });
                  })
                  .catch(error => {
                    setSubmitting(false);
                    Swal.fire('Error', error, 'error');
                  });
              }}
            >
              {submitting ? <Spinner /> : 'Post'}
            </Button>
          </Form>
        </Col>
      </Row>
    </Container>
  );
};

export default PostCreate;
