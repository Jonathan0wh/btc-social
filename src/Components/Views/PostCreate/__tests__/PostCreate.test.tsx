import React from 'react';
import { shallow } from 'enzyme';
import PostCreate from '../PostCreate';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn()
  })
}));

describe('<PostCreate />', () => {
  it('should render by default', () => {
    const wrapper = shallow(<PostCreate />);
    expect(wrapper).toMatchSnapshot();
  });
});
