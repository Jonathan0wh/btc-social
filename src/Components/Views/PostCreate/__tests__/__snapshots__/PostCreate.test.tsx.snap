// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCreate /> should render by default 1`] = `
<Container
  className="container"
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <h3>
        Create a Post
      </h3>
      <Form
        tag="form"
      >
        <FormGroup
          tag="div"
        >
          <Label
            for="post_title"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Title
          </Label>
          <Input
            id="post_title"
            onChange={[Function]}
            type="text"
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Label
            for="post_body"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Content
          </Label>
          <Input
            className="textarea"
            id="post_body"
            onChange={[Function]}
            type="textarea"
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Label
            for="post_userid"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            My user ID
          </Label>
          <Input
            id="post_userid"
            onChange={[Function]}
            type="number"
          />
        </FormGroup>
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
        >
          Post
        </Button>
      </Form>
    </Col>
  </Row>
</Container>
`;
